vllm_spyre/compilation_utils.py

```diff
@@ -0,0 +1,228 @@
+import json
+import os
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+# Third Party
+from vllm.logger import init_logger
+
+# Local
+import vllm_spyre.envs as envs_spyre
+
+if TYPE_CHECKING:
+    from vllm.config import ModelConfig, VllmConfig
+else:
+    ModelConfig = None
+    VllmConfig = None
+
+logger = init_logger(__name__)
+
+PRE_COMPILE_MODEL_CONFIG_FILENAME = "model_compile.log.json"
+PRE_COMPILE_MODEL_CATALOG_FILENAME = "pre_compiled_cache_catalog.json"
+DISABLE_COMPILATION_ENV_VAR = "DISABLE_COMPILATION"
+
+
+def handle_disable_compilation(vllm_config: VllmConfig, is_decoder: bool):
+    """
+    For decoder models we want to respect the `REQUIRE_PRECOMPILED_DECODERS`
+    environment variable, which disallows torch_sendnn from compiling new
+    graphs and only allows it to load pre-compiled graphs.
+    In order to do this, we must load up some config from the torch_sendnn
+    cache and check to make sure that the current vllm config matches,
+    otherwise the cached artifacts cannot be used.
+
+    For encoder models, we do not allow disabling compilation.
+    """
```

Comment:
```suggestion
    """
    The `DISABLE_COMPILATION` environment variable disallows torch_sendnn from
    compiling new graphs forcing it to load from the cache. Enabling
    `VLLM_SPYRE_REQUIRE_PRECOMPILED_DECODERS` will force DISABLE_COMPILATION
    for decoder models and require pre-compiled models.

    In order to do this, we must load up some config from the torch_sendnn
    cache and check to make sure that the current vllm config matches,
    otherwise the cached artifacts cannot be used.
    """
```